server/src/server.ts

```diff
@@ -382,17 +382,21 @@ function parseFile(textDocUri: string, docContent: string) {
     // Reset all the document's completion items.
     completionProvider.removeDocumentCompletionItems(textDocUri);
 
+    // There are syntax errors. Send the errors, if required, to the client and don't visit the failing files.
```

Comment:
In fact, the process was:
1. parse the file and generate error messages to display, if any and required;
1. visit the parsing result and create completion items, whatever the result of the parsing.

Now I propose:

1. parse the file and generate error messages to display, if any and required;
1. if no parsing error, visit the parsing result and create completion items.

If you think this will break too many things, I can remove this commit and postpone another solution.

Reply:
? So if, in a long file, there’s a tiny mistake (or some exotic YML feature that your grammar cannot yet understand), the ser won’t get any suggestion from this file? :thinking:

Reply:
Hum… yeah.
You're right, I remove this commit and will think about a better solution (I have some leads).

Reply:
Removed :+1:

Reply:
Great :-) Thanks @mminot-yseop for sharing my opinion :-)